undofx/src/main/java/org/fxmisc/undo/impl/UndoManagerImpl.java

```diff
@@ -209,7 +229,7 @@ private void addChange(C change) {
     @SuppressWarnings("unchecked")
     private C[] merge(C c1, C c2) {
         Optional<C> merged = merge.apply(c1, c2);
-        if(merged.isPresent()) {
+        if(merged.isPresent() && isNotIdentity.test(merged.get())) {
             return (C[]) new Object[] { merged.get() };
         } else {
             return (C[]) new Object[] { c1, c2 };
```

Comment:
When the change stream emits a non-identity change, `addChange` is called, right? (I've in-lined the `merge` method below:
````java
@SuppressWarnings("unchecked")
private void addChange(C change) {
    if(canMerge && queue.hasPrev()) {
        C prev = queue.prev();
        // merge in-lined start
        Optional<C> merged = merge.apply(prev, change);
        C[] changeArray;
        if(merged.isPresent() && isNotIdentity.test(merged.get())) {
            changeArray = (C[]) new Object[] { merged.get() };
        } else {
            changeArray = (C[]) new Object[] { c1, c2 };
        }
        // merge in-lined end
        queue.push(changeArray);
    } else {
        queue.push(change);
    }
    canMerge = true;
    undoAvailable.invalidate();
    redoAvailable.invalidate();
    atMarkedPosition.invalidate();
}
````
If I return an empty array, won't that mean the emitted change is not stored at all?